bokeh/util/compiler.py

```diff
@@ -85,22 +85,41 @@ def __str__(self):
 
 bokehjs_dir = settings.bokehjsdir()
 
-def nodejs_compile(code, lang="javascript", file=None):
-    compilejs_script = join(bokehjs_dir, "js", "compile.js")
+def _detect_nodejs():
+    if settings.nodejs_path() is not None:
+        nodejs_paths = [settings.nodejs_path()]
+    else:
+        nodejs_paths = ["xnodejs", "node"]
```

Comment:
Yep, it's a NASA version developed for environments with higher levels of background radiation, aka. I'm committing garbage.